src/basic/os-util.c

```diff
@@ -58,7 +58,47 @@ int path_is_extension_tree(const char *path, const char *extension) {
         return 1;
 }
 
-int open_extension_release(const char *root, const char *extension, char **ret_path, int *ret_fd) {
+static int check_extension_release_xattr(int extension_release_fd, char *extension_release_dir_path, char *filename) {
+        int r;
+
+        assert(extension_release_fd >= 0);
+        assert(extension_release_dir_path);
+        assert(filename);
+
+        /* No xattr or cannot parse it? Then skip this. */
+        _cleanup_free_ char *extension_release_xattr = NULL;
+        r = fgetxattr_malloc(extension_release_fd, "user.extension-release.strict", &extension_release_xattr);
+        if (r < 0) {
+                if (!ERRNO_IS_XATTR_ABSENT(r))
+                        log_debug_errno(r,
+                                        "%s/%s: Failed to read 'user.extension-release.strict' extended attribute from file, ignoring: %m",
+                                        extension_release_dir_path, filename);
+                else
+                        log_debug("%s/%s does not have user.extension-release.strict xattr, ignoring.", extension_release_dir_path, filename);
+
+                return r;
```

Comment:
Then, please return `false` here (or `true`??).

Reply:
Not addressed yet?

Reply:
this should not return true or false, it should return the enoent error as-is

Reply:
So, the failure here is also critical? Then, I'd like to drop ", ignoring" from the log message.

Reply:
"ignoring" here refers to the extension-release file - it will be ignored because it does not have the attribute

Reply:
OK. Then, how about to explicitly say `"ignoring the extension-release file"`?

Reply:
it already adds the full path and filename as the first thing in the debug message

Reply:
Yeah, I know. But still I'd like that.

But, not a strong opinion. I just confused when I read the code.

Reply:
because true means it's set and true, enoent means it's not set at all

Reply:
I'm with Yu on this… Saying "ignoring" and then returning error in some cases, and positive in another feels wrong. Overall, I think it'd be better if this function didn't say "ignoring" at all, since that's up to the caller.

Reply:
This is already very complicated for my users, and a major source of support requests. I do not want to change the log messages, as that will make things even more confusing. This is just a static helper, if the log messages don't fit with it, I'll merge it back into the caller.

Reply:
Makes sense.